[Platform security] Replace SCSS with CSS in JS (part 2)

SiddharthMantri opened the pull request:
Closes https://github.com/elastic/kibana/issues/211652

## Summary

Part 2 of https://github.com/elastic/kibana/issues/211652 and follow up from https://github.com/elastic/kibana/pull/214798

Removed most SASS files from all platform security owned code. (Full checklist on parent issue)

### How to test
Testing visual regression isn't super straightforward here. For my local testing, i started two instances of Kibana (main and this branch)

On main:

- Start es:
```
yarn es snapshot --license=trial -E http.port=9400
```

- Start kibana with the following config (CLI or kibana.dev.yml)
``` 
server.port: 5602
elasticsearch.hosts: ["http://localhost:9400"]
```
Once started, in a private browsing window, you should have access to Kibana on main on `localhost:5602`

On this PR:
Start ES and Kibana normally (Kibana should be available on localhost:5601)

This PR contains changes to the following parts of the Spaces plugin:
- Access agreement page
- Login page (see note below regarding testing)
- Role mapping rule group editor

For testing the full login page with different selectors, etc, you can start kibana with the following changes in the kibana.dev.yml file
```
xpack.security.authc.providers:
  basic.basic1:
    order: 0
    hint: "Use your corporate username & password"
    accessAgreement:
      message: |
        **You are accessing a system with sensitive information**

        By logging in, you acknowledge that information system usage

  pki.pki1:
    order: 1

  token.token1:
    order: 2

  saml.saml1:
    order: 3
    realm: saml1

  oidc.oidc1:
    order: 4
    realm: oidc1
```

Ideally, you should see no visual regression between the two versions. 


## Screenshots

| Component | Main | PR |
|--------|--------|--------|
|Access agreement page|<img width="300" alt="Screenshot 2025-05-09 at 10 31 18" src="https://github.com/user-attachments/assets/483d9e47-e4a0-4e01-8282-9d205ba0bf6d" />|<img width="300" alt="Screenshot 2025-05-09 at 10 54 14" src="https://github.com/user-attachments/assets/62965843-5242-4d96-b300-812cdc349864" />|
|Login page|<img width="300" alt="Screenshot 2025-05-08 at 17 05 23" src="https://github.com/user-attachments/assets/99536f62-75d8-46e2-ae3a-aba7b990a723" />|<img width="300" alt="Screenshot 2025-05-08 at 18 24 50" src="https://github.com/user-attachments/assets/3c2aadc9-1746-4d7f-9093-a6053916b597" />|
|Role mapping rule editor| <img width="300" src="https://github.com/user-attachments/assets/8a14e323-223d-43df-9dfb-edf8f6859c40" />|<img width="300" src="https://github.com/user-attachments/assets/68a9b37d-060d-4060-b2ad-b0b3319de78d" />|


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [x] The risk of inexact conversion: verifying this PR requires manual checks to ensure that the conversion has not created any regressions in the style.



<!--ONMERGE {"backportTargets":["8.18","8.19"]} ONMERGE-->

SiddharthMantri committed:
[Platform security] Replace SCSS with CSS in JS (part 2)

SiddharthMantri committed:
fix prop name

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/334198)
* Commit: 51d85597429d05fe154e0f99a7ca79ab4f8952ba

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/334198#0198f693-3263-4fb0-b958-c7374e515a41)
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/334198#0198f6af-54ff-41c0-8fd2-3ffc75ca988b)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334198","number":334198,"commit":"51d85597429d05fe154e0f99a7ca79ab4f8952ba"}],"number":334198}
buildkite-pr-comment-->

SiddharthMantri committed:
update snapshots

SiddharthMantri committed:
update snaps

SiddharthMantri committed:
fix app.tsx file

SiddharthMantri committed:
Merge branch 'main' into 211652-security-scss-emotion-p2

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335678)
* Commit: 55367ee4256a96f351608806536259c5f5b195a2

### Failed CI Steps
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/335678#01991417-e6c8-4bde-a231-c401b571825f)
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/335678#01991436-e2ce-4b98-b611-d708d7fa3ab8)

### History
* :broken_heart: [Build #334249](https://buildkite.com/elastic/kibana-pull-request/builds/334249) failed 9a4e4f281504fc060b51b9adaea961b6a77c59a2
* :broken_heart: [Build #334198](https://buildkite.com/elastic/kibana-pull-request/builds/334198) failed 51d85597429d05fe154e0f99a7ca79ab4f8952ba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335678","number":335678,"commit":"55367ee4256a96f351608806536259c5f5b195a2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334249","number":334249,"commit":"9a4e4f281504fc060b51b9adaea961b6a77c59a2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334198","number":334198,"commit":"51d85597429d05fe154e0f99a7ca79ab4f8952ba"}],"number":335678}
buildkite-pr-comment-->

elasticmachine committed:
Merge branch 'main' into 211652-security-scss-emotion-p2

elasticmachine committed:
Merge branch 'main' into 211652-security-scss-emotion-p2

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343510)
* Commit: 0d68f22594a05f4380e48c20507df9b40d01ce18

### Failed CI Steps
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/343510#01997fbb-7082-487e-a42e-e72c9fec8d0c)
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/343510#01997fcf-991b-4a4e-b678-a58f7d553b86)

### History
* :broken_heart: [Build #335678](https://buildkite.com/elastic/kibana-pull-request/builds/335678) failed 55367ee4256a96f351608806536259c5f5b195a2
* :broken_heart: [Build #334249](https://buildkite.com/elastic/kibana-pull-request/builds/334249) failed 9a4e4f281504fc060b51b9adaea961b6a77c59a2
* :broken_heart: [Build #334198](https://buildkite.com/elastic/kibana-pull-request/builds/334198) failed 51d85597429d05fe154e0f99a7ca79ab4f8952ba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343510","number":343510,"commit":"0d68f22594a05f4380e48c20507df9b40d01ce18"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335678","number":335678,"commit":"55367ee4256a96f351608806536259c5f5b195a2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334249","number":334249,"commit":"9a4e4f281504fc060b51b9adaea961b6a77c59a2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334198","number":334198,"commit":"51d85597429d05fe154e0f99a7ca79ab4f8952ba"}],"number":343510}
buildkite-pr-comment-->

SiddharthMantri committed:
update snapshots

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343578)
* Commit: ca395bbf7ce2962f2990be1c1d2b94c65225a72e

### Failed CI Steps
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e)
* [Jest Tests #14](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffea-4e8b-a577-8b42c1e45e23) Jest Tests #<span></span>14 / <SimplePrivilegeForm> allows feature privileges to be customized
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9605-405e-b567-370e36fdd441) Jest Tests #<span></span>14 / <SimplePrivilegeForm> allows feature privileges to be customized
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffeb-481d-908c-2d4d75014872) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays "custom" when feature privileges are customized
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9606-4f16-8a0f-effbfa832a2a) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays "custom" when feature privileges are customized
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffea-4137-98a8-65a467083e37) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays "none" when no privilege is selected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-960c-4d53-a7e6-e070e3167753) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays "none" when no privilege is selected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998058-0027-4f69-9740-4f7f1f89b358) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays the reserved privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-960a-499f-b3ff-0369076f18ab) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays the reserved privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffe7-44b8-8978-df6c0b7fdcab) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays the selected privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-960e-4659-8dd0-4744e230acfa) Jest Tests #<span></span>14 / <SimplePrivilegeForm> displays the selected privilege
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffed-47c7-aad8-d1dfb7cada83) Jest Tests #<span></span>14 / <SimplePrivilegeForm> renders without crashing
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9605-4f06-806f-74a2df6655c4) Jest Tests #<span></span>14 / <SimplePrivilegeForm> renders without crashing
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffea-4a8e-b7c8-8aecea4bb372) Jest Tests #<span></span>14 / AuthenticationStatePage renders the title, child content, and layout elements correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9609-44ba-912f-c928c416644f) Jest Tests #<span></span>14 / AuthenticationStatePage renders the title, child content, and layout elements correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffe8-41df-a4b0-6b61d3dc8aea) Jest Tests #<span></span>14 / LoginForm renders as expected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-966c-4e54-a025-29cc4df83c1f) Jest Tests #<span></span>14 / LoginForm renders as expected
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffe7-454b-b311-e2a109403de9) Jest Tests #<span></span>14 / RuleGroupEditor allows the group type to be changed, maintaining child rules
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9607-4a18-8e67-6324056710ee) Jest Tests #<span></span>14 / RuleGroupEditor allows the group type to be changed, maintaining child rules
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998058-0026-4094-8005-939637bdf67d) Jest Tests #<span></span>14 / RuleGroupEditor can render a readonly view
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-9609-4281-8982-3ab83ad94451) Jest Tests #<span></span>14 / RuleGroupEditor can render a readonly view
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffea-44d7-b3a7-7b4867d29463) Jest Tests #<span></span>14 / RuleGroupEditor does not change groups when canceling the confirmation
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-960c-4b39-8850-318c2f5650bd) Jest Tests #<span></span>14 / RuleGroupEditor does not change groups when canceling the confirmation
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998034-1b6b-49ca-93f7-87ee306c386e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998034-1b6b-49ca-93f7-87ee306c386e/artifacts/01998057-ffea-4be9-9701-4b06a49a4bf6) Jest Tests #<span></span>14 / RuleGroupEditor warns when changing group types which would invalidate child rules
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343578#01998058-125f-4fb6-8675-0e8cb3e91ab8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343578/jobs/01998058-125f-4fb6-8675-0e8cb3e91ab8/artifacts/01998064-966c-4382-b7ac-2857cbbb6064) Jest Tests #<span></span>14 / RuleGroupEditor warns when changing group types which would invalidate child rules

### History
* :broken_heart: [Build #343510](https://buildkite.com/elastic/kibana-pull-request/builds/343510) failed 0d68f22594a05f4380e48c20507df9b40d01ce18
* :broken_heart: [Build #335678](https://buildkite.com/elastic/kibana-pull-request/builds/335678) failed 55367ee4256a96f351608806536259c5f5b195a2
* :broken_heart: [Build #334249](https://buildkite.com/elastic/kibana-pull-request/builds/334249) failed 9a4e4f281504fc060b51b9adaea961b6a77c59a2
* :broken_heart: [Build #334198](https://buildkite.com/elastic/kibana-pull-request/builds/334198) failed 51d85597429d05fe154e0f99a7ca79ab4f8952ba

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343578","number":343578,"commit":"ca395bbf7ce2962f2990be1c1d2b94c65225a72e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343510","number":343510,"commit":"0d68f22594a05f4380e48c20507df9b40d01ce18"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335678","number":335678,"commit":"55367ee4256a96f351608806536259c5f5b195a2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334249","number":334249,"commit":"9a4e4f281504fc060b51b9adaea961b6a77c59a2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334198","number":334198,"commit":"51d85597429d05fe154e0f99a7ca79ab4f8952ba"}],"number":343578}
buildkite-pr-comment-->

elasticmachine committed:
Merge branch 'main' into 211652-security-scss-emotion-p2

SiddharthMantri committed:
test updates

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/343801)
* Commit: b011010961f247294d8cb90d509f960aaff4adfa

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/343801#0199816e-e3cd-459f-8c16-8e4b4b57778f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/343801#0199816e-e3cd-459f-8c16-8e4b4b57778f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/343801/jobs/0199816e-e3cd-459f-8c16-8e4b4b57778f/artifacts/01998190-5c05-4c2f-8f9b-d456dd77430b) Scout: [ platform / streams_app ] plugin / stateful - Stream data routing - error handling and recovery - should handle network failures during rule creation

### History
* :broken_heart: [Build #343578](https://buildkite.com/elastic/kibana-pull-request/builds/343578) failed ca395bbf7ce2962f2990be1c1d2b94c65225a72e
* :broken_heart: [Build #343510](https://buildkite.com/elastic/kibana-pull-request/builds/343510) failed 0d68f22594a05f4380e48c20507df9b40d01ce18
* :broken_heart: [Build #335678](https://buildkite.com/elastic/kibana-pull-request/builds/335678) failed 55367ee4256a96f351608806536259c5f5b195a2
* :broken_heart: [Build #334249](https://buildkite.com/elastic/kibana-pull-request/builds/334249) failed 9a4e4f281504fc060b51b9adaea961b6a77c59a2
* :broken_heart: [Build #334198](https://buildkite.com/elastic/kibana-pull-request/builds/334198) failed 51d85597429d05fe154e0f99a7ca79ab4f8952ba

cc @kc13greiner

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343801","number":343801,"commit":"b011010961f247294d8cb90d509f960aaff4adfa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343578","number":343578,"commit":"ca395bbf7ce2962f2990be1c1d2b94c65225a72e"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/343510","number":343510,"commit":"0d68f22594a05f4380e48c20507df9b40d01ce18"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335678","number":335678,"commit":"55367ee4256a96f351608806536259c5f5b195a2"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334249","number":334249,"commit":"9a4e4f281504fc060b51b9adaea961b6a77c59a2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/334198","number":334198,"commit":"51d85597429d05fe154e0f99a7ca79ab4f8952ba"}],"number":343801}
buildkite-pr-comment-->

kc13greiner committed:
Update rule_group_editor.test.tsx

Now that <EuiProvider> is the wrapper, this expect is finding the passed in RuleGroupEditor.

No sub-RuleGroupEditors are found though.

We need mount vs shallow here so we can find the AddRuleButton

kc13greiner commented on x-pack/platform/plugins/shared/security/public/management/role_mappings/edit_role_mapping/rule_editor_panel/rule_group_editor.test.tsx:
```diff
@@ -25,8 +31,8 @@ describe('RuleGroupEditor', () => {
     };
-    const wrapper = shallowWithIntl(<RuleGroupEditor {...props} />);
-    expect(wrapper.find(RuleGroupEditor)).toHaveLength(0);
+    const wrapper = mountWithEuiProvider(<RuleGroupEditor {...props} />);
```
Note: now that it is nested in the EuiProvider, the first assertion is finding the element that was passed in. 

So just one is found, no sub RuleGroupEditors

kc13greiner committed:
Update simple_privilege_section.test.tsx

Now that we are actually rendering it with mount (vs shallow), it needs the `role` prop set

legrego committed:
Merge branch 'main' into 211652-security-scss-emotion-p2

kc13greiner reviewed (approved):
LGTM!
